clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -848,6 +848,8 @@ public void handleResponse(AbstractResponse response) {
                         coordinatorReloaded = true;
                         lookupCoordinator(FindCoordinatorRequest.CoordinatorType.GROUP, builder.consumerGroupId());
                     }
+                } else if (error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
```

Comment:
Could we check that the `response.requestHeader.apiKey` is either `addPartitionsToTxn` or `txnCommitOffset`? For example `addOffsetsToTxn` should not return this error code and hence if yes we should fail fast as a bug.